src/overlays/actors/ovl_En_Sc_Ruppe/z_en_sc_ruppe.c

```diff
@@ -31,29 +37,155 @@ const ActorInit En_Sc_Ruppe_InitVars = {
+    { gameplay_keep_Tex_062020, 50 },  // Purple rupee
+    { gameplay_keep_Tex_062060, 10 },  // Gray rupee (unused)
 };
 
-#endif
-
-extern ColliderCylinderInit D_80BD6E40;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/func_80BD6910.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/func_80BD697C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/func_80BD6A8C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/func_80BD6B18.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/EnScRuppe_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/EnScRuppe_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/EnScRuppe_Update.s")
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_NO_PUSH | OC1_TYPE_PLAYER,
+        OC2_TYPE_2,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK4,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 10, 30, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sc_Ruppe/EnScRuppe_Draw.s")
+void func_80BD6910(EnScRuppe* this, GlobalContext* globalCtx) {
+    Collider_UpdateCylinder(&this->actor, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, this, 32.0f, 30.0f, 0.0f, 4U);
+}
+
+s32 func_80BD697C(s16 this) {
+    switch (this) {
+        case 0:
+            if (gSaveContext.save.weekEventReg[53] & 4) {
+                gSaveContext.save.weekEventReg[53] &= (u8)~4;
+                return 1;
+            }
+            break;
+        case 1:
+            if (gSaveContext.save.weekEventReg[53] & 0x80) {
+                gSaveContext.save.weekEventReg[53] &= (u8)~0x80;
+                return 1;
+            }
+            break;
+        case 2:
+            if (gSaveContext.save.weekEventReg[54] & 1) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~1;
+                return 1;
+            }
+            break;
+        case 3:
+            if (gSaveContext.save.weekEventReg[54] & 2) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~2;
+                return 1;
+            }
+            break;
+        case 4:
+            if (gSaveContext.save.weekEventReg[54] & 4) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~4;
+                return 1;
+            }
+            break;
+        case 5:
+            if ((gSaveContext.save.weekEventReg[54] & 8)) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~8;
+                return 1;
+            }
+            break;
+    }
+    return 0;
+}
+
+void func_80BD6A8C(EnScRuppe* this, GlobalContext* globalCtx) {
+    if (this->collider.base.ocFlags1 & 2) {
```

Comment:
```suggestion
    if (this->collider.base.ocFlags1 & OC1_HIT) {
```
Replace with the collision check define